cfspopcon/input_file_handling.py

```diff
@@ -13,7 +13,7 @@
 
 
 def read_case(
-    case: Union[str, Path], kwargs: Optional[dict[str, str]] = None
+    case: Union[str, Path, dict[str, Any]], kwargs: Optional[dict[str, str]] = None
 ) -> tuple[dict[str, Any], Union[CompositeAlgorithm, Algorithm], dict[str, Any], dict[str, Path]]:
     """Read a yaml file corresponding to a given case.
```

Comment:
How do you feel about leaving `read_case` targeted at reading a yaml file?

If one wants to do things programmatically, one could directly use the `process_*` functions, right? And then one also doesn't have to first create one large dict with all config, to pass it into `read_case`.

I've got an example notebook that does something like that, which I once created to show that workflow to another cfspopcon user. I'd be happy to update it a bit to have it as an example in the docs.

Reply:
I've split `process_input_dictionary` from `read_case` — so now can choose to use the yaml read and processing, or to just apply the processing.